src/RemoteHandler/ServerMock.js

```diff
@@ -8,22 +8,132 @@ class ServerMock {
+                    id: 0,
+                    causeDescription: "Kabelbaum missing",
+                    errorOccurrences: 3,
+                    errorPrediction: 0.3
+                }]
+            }]
+        },
+        {
+            id: 200,
+            productId: 0,
+            state: "WARNING",
+            stateDescription: "Few errors or warnings",
+            codingPlugId: 20,
+            codingPlugSoftwareVersions: [2.0, 2.1],
+            numberOfEvents: 234,
+            statusUpdateQueries: {
+                RUNNING: "ESPER-QUERY",
+                WARNING: "ESPER-QUERY",
+                ERROR: "ESPER-QUERY"
+            },
+            errorTypes: [{
+                id: 0,
+                errorTypeId: "0",
+                displayCode: "0Z",
+                causeCode:  0,
+                errorDescription: "Heating produces strange sounds",
+                errorCauses: [{
+                    id: 0,
+                    causeDescription: "Kabelbaum missing",
+                    errorOccurrences: 3,
+                    errorPrediction: 0.3
+                }]
+            }]
+        },{
+            id: 300,
+            productId: 0,
+            state: "ERROR",
+            stateDescription: "Too many errors",
+            codingPlugId: 30,
+            codingPlugSoftwareVersions: [1.0, 1.1, 3.1],
+            numberOfEvents: 234,
+            statusUpdateQueries: {
+                RUNNING: "ESPER-QUERY",
+                WARNING: "ESPER-QUERY",
+                ERROR: "ESPER-QUERY"
+            },
+            errorTypes: [{
+                id: 0,
+                errorTypeId: "0",
+                displayCode: "0Z",
+                causeCode:  0,
+                errorDescription: "Heating produces strange sounds",
+                errorCauses: [{
+                    id: 0,
+                    causeDescription: "Kabelbaum missing",
+                    errorOccurrences: 3,
+                    errorPrediction: 0.3
+                }]
+            }]
+        },{
+            id: 400,
+            productId: 0,
+            state: "UNDEFINED",
+            stateDescription: "No status information yet.",
+            codingPlugId: 10,
+            codingPlugSoftwareVersions: [4.0, 4.1],
+            numberOfEvents: 234,
+            statusUpdateQueries: {
+                RUNNING: "ESPER-QUERY",
+                WARNING: "ESPER-QUERY",
+                ERROR: "ESPER-QUERY"
+            },
+            errorTypes: [{
+                id: 0,
+                errorTypeId: "0",
+                displayCode: "0Z",
+                causeCode:  0,
+                errorDescription: "Heating produces strange sounds",
+                errorCauses: [{
+                    id: 0,
+                    causeDescription: "Kabelbaum missing",
+                    errorOccurrences: 3,
+                    errorPrediction: 0.3
+                }]
+            }]
+        }];
+    }
+
     static getProductFamily() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') This function has 199 lines, which is greater than the 100 lines authorized. Split it into smaller functions. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS138)